src/tabs/tabs.tsx

```diff
@@ -207,16 +209,39 @@ function TabList({
                     width: `${selectedTab.offsetWidth}px`,
                 })
             }
-        }
+        },
+        [selectedId],
+    )
+
+    React.useEffect(
+        function updateSelectedTabPositionOnTabChange() {
+            updateSelectedTabPosition()
+        },
+        // `selectedId` is a dependency to ensure the effect runs when the selected tab changes
+        [selectedId, updateSelectedTabPosition],
+    )
 
-        updateSelectedTabStyle()
+    React.useEffect(
+        function observeTabListWidthChange() {
+            const tabList = new ResizeObserver(([entry]) => {
```

Comment:
I think it's still worth having this be throttled (rather than debounced if we want to see transitional frames), for somewhere between 64ms (so it's capped at ~30fps, to 100ms for 10fps).

Also slight nit:

```suggestion
            const tabListObserver = new ResizeObserver(([entry]) => {
```

Reply:
The problem with "throttling" is that we don't have lodash on Reactist, and I'd have to implement a custom function just for this (or copy some battle-tested throttle function from somewhere). Alternative, use lodash and add it as a peer dependency to Reactist. Any other options? How do you think we should handle this?

Reply:
How about using [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame)? Feels like the most appropriate solution. I've pushed a commit with that change, let me know what you think.